src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/dto/converter/CaInfoConverter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019- Nordic Institute for Interoperability Solutions (NIIS)
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ * <p>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * <p>
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * <p>
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.niis.xroad.centralserver.restapi.dto.converter;
+
+import ee.ria.xroad.common.util.CertUtils;
+import ee.ria.xroad.common.util.CryptoUtils;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.niis.xroad.centralserver.restapi.dto.CertificateDetails;
+import org.niis.xroad.centralserver.restapi.entity.CaInfo;
+import org.springframework.stereotype.Component;
+
+import java.security.PublicKey;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPublicKey;
+
+import static ee.ria.xroad.common.util.CertUtils.getIssuerCommonName;
+import static ee.ria.xroad.common.util.CertUtils.getSubjectAlternativeNames;
+import static ee.ria.xroad.common.util.CertUtils.getSubjectCommonName;
+import static java.lang.String.valueOf;
+
+@Component
+@RequiredArgsConstructor
+public class CaInfoConverter {
+
+    private final KeyUsageConverter keyUsageConverter;
+
+    @SneakyThrows
+    public CertificateDetails convert(CaInfo caInfo) {
```

Comment:
was use of mapstruct considered here?  I do see pros and cons of using it so curious regarding this decision

Reply:
Yes, it was. But when mapping from entity to dto (dto fields naming is based on openapi dto), many fields would be mapped using annotations. The amount of code lines would be similar, but it is more error prone.